python/sglang/srt/layers/attention/trtllm_mla_backend.py

```diff
@@ -0,0 +1,370 @@
+        kv_indices_buf: Optional[torch.Tensor] = None,
+    ):
+        """Initialize CUDA graph state for TRTLLM MLA."""
+        max_blocks_per_seq = self._calc_padded_blocks(self.max_context_len)
+
+        self.cuda_graph_kv_indices = torch.full(
+            (max_bs, max_blocks_per_seq), -1, dtype=torch.int32, device=self.device
+        )
+        self.cuda_graph_workspace = torch.empty(
+            self.workspace_size, dtype=torch.int8, device=self.device
+        )
+
+    def init_forward_metadata_capture_cuda_graph(
+        self,
+        bs: int,
+        num_tokens: int,
+        req_pool_indices: torch.Tensor,
+        seq_lens: torch.Tensor,
+        encoder_lens: Optional[torch.Tensor],
+        forward_mode: ForwardMode,
+        spec_info: Optional[SpecInfo],
+    ):
+        """Initialize metadata for CUDA graph capture."""
+        if forward_mode.is_decode_or_idle() and spec_info is None:
+            max_seqlen_pad = self._calc_padded_blocks(seq_lens.max().item())
+            block_kv_indices = self.cuda_graph_kv_indices[:bs, :max_seqlen_pad]
+
+            create_flashmla_kv_indices_triton[(bs,)](
+                self.req_to_token,
+                req_pool_indices,
+                seq_lens,
+                None,
+                block_kv_indices,
+                self.req_to_token.stride(0),
+                max_seqlen_pad,
+                self.page_size,
+            )
+
+            metadata = TRTLLMMLADecodeMetadata(
+                self.cuda_graph_workspace, block_kv_indices
+            )
+            self.decode_cuda_graph_metadata[bs] = metadata
+            self.forward_metadata = metadata
+        else:
+            super().init_forward_metadata_capture_cuda_graph(
+                bs,
+                num_tokens,
+                req_pool_indices,
+                seq_lens,
+                encoder_lens,
+                forward_mode,
+                spec_info,
+            )
+
+    def init_forward_metadata_replay_cuda_graph(
+        self,
+        bs: int,
+        req_pool_indices: torch.Tensor,
+        seq_lens: torch.Tensor,
+        seq_lens_sum: int,
+        encoder_lens: Optional[torch.Tensor],
+        forward_mode: ForwardMode,
+        spec_info: Optional[SpecInfo],
+        seq_lens_cpu: Optional[torch.Tensor],
+    ):
+        """Replay CUDA graph with new inputs."""
+        if forward_mode.is_decode_or_idle() and spec_info is None:
+            metadata = self.decode_cuda_graph_metadata[bs]
+
+            # Update block indices for new sequences
+            create_flashmla_kv_indices_triton[(bs,)](
+                self.req_to_token,
+                req_pool_indices[:bs],
+                seq_lens[:bs],
+                None,
+                metadata.block_kv_indices,
+                self.req_to_token.stride(0),
+                metadata.block_kv_indices.shape[1],
+                self.page_size,
+            )
+
+            self.forward_metadata = metadata
```

Comment:
nit: is this line redundant, since if we replay cuda graph, our forward pass will not touch this variable (but instead directly run kernels in cuda graph)

for safety, I even consider explicitly del this field